sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -65,7 +65,8 @@ case class AdaptiveSparkPlanExec(
     inputPlan: SparkPlan,
     @transient context: AdaptiveExecutionContext,
     @transient preprocessingRules: Seq[Rule[SparkPlan]],
-    @transient isSubquery: Boolean)
+    @transient isSubquery: Boolean,
+    @transient outputColumnar: Boolean = false)
```

Comment:
I like the idea of adding this parameter but I'm not understanding how this would get set to `true`, especially from a Spark plugin.  It looks like new instances of `AdaptiveSparkPlanExec` are only created by the `InsertAdaptiveSparkPlan` rule and that does not explicitly specify a value for `outputColumnar` in this PR. Would a plugin need to replace this rule with a custom version perhaps?

Reply:
For example:
```
case class WriteToParquetColumnarExec(input: SparkPlan, ...) extends UnaryExecNode {
  def doExecute... = {
    val finalPlan = input match {
      case a: AdaptiveSparkPlanExec => a.copy(outputColumnar = true)
      case _ => input
    }
    finalPlan.executeColumnar().mapPartitions ...
  }
}
```

Another idea is to add custom physical rules. Currently, we don't have an extension point to allow people to add custom physical rules (that deal with the physical plan produced by the planner). One workaround is to leverage `ColumnarRule.postColumnarTransitions`:
```
class MyColumnarRule... {
  def postColumnarTransitions = new Rule[SparkPlan] {
    def apply(input: SparkPlan) = input match {
      case w @ WriteToParquetColumnarExec(a: AdaptiveSparkPlanExec, ...) =>
        w.copy(input = a.copy(outputColumnar = true))
      case other => other
    }
  }
}
```

Reply:
Thanks. I really like this approach and this will work for us.